caas/kubernetes/provider/constraints.go

```diff
@@ -23,5 +27,80 @@ var unsupportedConstraints = []string{
 func (k *kubernetesClient) ConstraintsValidator(ctx context.ProviderCallContext) (constraints.Validator, error) {
 	validator := constraints.NewValidator()
 	validator.RegisterUnsupported(unsupportedConstraints)
-	return validator, nil
+	return &constraintsValidator{Validator: validator}, nil
+}
+
+type constraintsValidator struct {
+	constraints.Validator
+}
+
+func validateTopologyKeyTagFunc(key string, topologySpreadKey string, tagList []string) error {
+	var topologySpreadValue, otherValue *string
+	// Search for the topologySpreadValue for topologyKeyTag
+	for tag := range tagList {
+		if strings.HasPrefix(tagList[tag], topologySpreadKey) {
+			trimmedValue := strings.TrimPrefix(tagList[tag], topologySpreadKey+"=")
+			topologySpreadValue = &trimmedValue
+			break
+		}
+	}
+	if topologySpreadValue == nil {
+		// Nothing to do, continue to the next check
+		return nil
+	}
+
+	for tag := range tagList {
+		if strings.HasPrefix(tagList[tag], key) {
+			trimmedValue := strings.TrimPrefix(tagList[tag], key+"=")
+			otherValue = &trimmedValue
+			break
+		}
+	}
+
+	if otherValue != nil && *topologySpreadValue == *otherValue {
+		return errors.New("podAffinity/antiPodAffinity's topology-key and topologySpread's cannot have the same value")
```

Comment:
`If both are specified, the topology-key for (anti) pod affinity and topology-spread cannot have the same value`